platform/viewer/src/connectedComponents/ConnectedHeader.js

```diff
@@ -1,10 +1,18 @@
 import Header from '../components/Header/Header.js';
 import { connect } from 'react-redux';
+import { hotkeysManager } from '../App.js';
+import cloneDeep from 'lodash.clonedeep';
 
 const mapStateToProps = state => {
+  const isEmpty = obj => Object.keys(obj).length === 0;
+  const newHotKeysData =
+    state.preferences && !isEmpty(state.preferences.hotKeysData)
+      ? state.preferences.hotKeysData
+      : hotkeysManager.hotkeyDefinitions;
+  hotkeysManager.setHotkeys(hotkeysManager.format(cloneDeep(newHotKeysData)));
```

Comment:
Instead of formatting/cloning here, can we offload that concern to the hotkeys manager?

Reply:
This was fixed with the latest changes. thx.